csvkit/utilities/csvstat.py

```diff
@@ -116,8 +117,8 @@ def add_arguments(self):
             '--count', dest='count_only', action='store_true',
             help='Only output total row count.')
         self.argparser.add_argument(
-            '-y', '--snifflimit', dest='sniff_limit', type=int,
-            help='Limit CSV dialect sniffing to the specified number of bytes. Specify "0" to disable sniffing.')
+            '-y', '--snifflimit', dest='sniff_limit', type=int, default=1024,
```

Comment:
Currently, `csvkit` functions do not use a sniff-limit by default. This feels problematic, as very large CSVs will spend huge runtimes to figure out the delimiter, which is inferable from a sample.

If we look at [python's documentation](https://docs.python.org/3/library/csv.html#csv.Sniffer), they call the string input a `sample` and use 1024 by default. I chose 1024 to align with this, but am okay changing the default!